fabric-networking-api-v1/src/client/java/net/fabricmc/fabric/mixin/networking/client/DisconnectedScreenMixin.java

```diff
@@ -0,0 +1,94 @@
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.networking.client;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+import org.checkerframework.common.aliasing.qual.Unique;
+import org.objectweb.asm.Opcodes;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.client.font.MultilineText;
+import net.minecraft.client.gui.screen.DisconnectedScreen;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.util.math.MathHelper;
+
+import net.fabricmc.fabric.impl.networking.client.DisconnectedScreenHooks;
+
+/**
+ * This mixin makes disconnect reason text scrollable.
+ */
+@Mixin(DisconnectedScreen.class)
+public abstract class DisconnectedScreenMixin extends Screen implements DisconnectedScreenHooks {
+	@Shadow
+	private int reasonHeight;
+
+	@Shadow
+	private MultilineText reasonFormatted;
+
+	@Unique
+	private int actualReasonHeight;
+
+	@Unique
+	private int scroll;
+
+	private DisconnectedScreenMixin() {
+		super(null);
+	}
+
+	// Inject to right after reasonHeight is stored, to make sure the back button have correct position.
+	@Inject(method = "init", at = @At(value = "FIELD", target = "Lnet/minecraft/client/gui/screen/DisconnectedScreen;reasonHeight:I", opcode = Opcodes.PUTFIELD, shift = At.Shift.AFTER))
+	private void init(CallbackInfo ci) {
+		actualReasonHeight = reasonHeight;
+		reasonHeight = Math.min(reasonHeight, height - 100);
+		scroll = 0;
+	}
+
+	@Redirect(method = "render", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/font/MultilineText;drawCenterWithShadow(Lnet/minecraft/client/util/math/MatrixStack;II)I"))
+	private int render(MultilineText instance, MatrixStack matrixStack, int x, int y) {
+		double scale = client.getWindow().getScaleFactor();
+		RenderSystem.enableScissor(0, (int) (y * scale), (int) (width * scale), (int) (reasonHeight * scale));
+		instance.drawCenterWithShadow(matrixStack, x, y - scroll);
+		RenderSystem.disableScissor();
+
+		// Draw gradient at the top/bottom to indicate that the text is scrollable.
+		if (actualReasonHeight > reasonHeight) {
+			int startX = (width - reasonFormatted.getMaxWidth()) / 2;
+			int endX = (width + reasonFormatted.getMaxWidth()) / 2;
+
+			if (scroll > 0) {
+				fillGradient(matrixStack, startX, y, endX, y + 10, 0xFF000000, 0);
+			}
+
+			if (scroll < actualReasonHeight - reasonHeight) {
+				fillGradient(matrixStack, startX, y + reasonHeight - 10, endX, y + reasonHeight, 0, 0xFF000000);
+			}
+		}
+
+		return y + reasonHeight;
+	}
+
+	@Override
+	public void fabric_mouseScrolled(double mouseX, double mouseY, double amount) {
```

Comment:
If im not mistake you should just able to override onMouseScrolled here, and remove the need for the 2nd mixin + interface?

Reply:
Fixed my self as I had it cloned down for testing.

Reply:
Overriding `Screen`s method in the mixin is pretty much the same as `@Overwrite` so I avoided it.

Reply:
Not really, as its adding a new method, not overriding anything. Can revert my change if you think its best?